tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -668,3 +219,331 @@ def _run(self):
                 payload={"message": "%s complete" % self.__class__.__name__}
             )
         )
+
+
+def emit_event(self, resource, curr_value, msg, instance):
+    alert = {}
+    alert['source'] = NS.publisher_id
+    alert['pid'] = os.getpid()
+    alert['time_stamp'] = tendrl_now().isoformat()
+    alert['alert_type'] = 'status'
+    severity = "INFO"
+    if curr_value.lower() == "stopped":
+        severity = "CRITICAL"
+    alert['severity'] = severity
+    alert['resource'] = resource
+    alert['current_value'] = curr_value
+    alert['tags'] = dict(
+        plugin_instance=instance,
+        message=msg,
+        cluster_id=NS.tendrl_context.integration_id,
+        cluster_name=NS.tendrl_context.cluster_name,
+        sds_name=NS.tendrl_context.sds_name,
+        fqdn=socket.getfqdn()
+    )
+    alert['node_id'] = NS.node_context.node_id
+    if not NS.node_context.node_id:
+        return
+    Event(
+        Message(
+            "notice",
+            "alerting",
+            {'message': json.dumps(alert)}
+        )
+    )
+
+
+def sync_volumes(volumes, index, vol_options):
+    SYNC_TTL = int(NS.config.data.get("sync_interval", 10)) + 250
+    node_context = NS.node_context.load()
+    tag_list = list(node_context.tags)
+    # Raise alerts for volume state change.
+    cluster_provisioner = "provisioner/%s" % NS.tendrl_context.integration_id
+    if cluster_provisioner in tag_list:
+        try:
+            stored_volume_status = NS._int.client.read(
+                "clusters/%s/Volumes/%s/status" % (
+                    NS.tendrl_context.integration_id,
+                    volumes['volume%s.id' % index]
+                )
+            ).value
+            current_status = volumes['volume%s.status' % index]
+            if stored_volume_status != "" and \
+                current_status != stored_volume_status:
+                msg = "Status of volume: %s " + \
+                      "changed from %s to %s" % (
+                          volumes['volume%s.name' % index],
+                          stored_volume_status,
+                          current_status
+                      )
+                instance = "volume_%s" % volumes[
+                    'volume%s.name' % index
+                ]
+                emit_event(
+                    "volume_status",
+                    current_status,
+                    msg,
+                    instance
+                )
+        except etcd.EtcdKeyNotFound:
+            pass
+
+    rebalance_status = ""
+    if volumes['volume%s.type' % index].startswith("Distribute"):
+        status = rebal_stat.get_rebalance_status(
+            volumes['volume%s.name' % index]
+        )
+        if status:
+            rebalance_status = status.replace(" ", "_")
+        else:
+            rebalance_status = "not_started"
+
+    volume = NS.gluster.objects.Volume(
+        vol_id=volumes['volume%s.id' % index],
+        vol_type="arbiter"
```

Comment:
`get-state` reports arbiter volumes as `replicated` only. If arbiter_count > 0 we mark the type as `arbiter`